notifications_utils/template.py

```diff
@@ -253,8 +254,35 @@ def fragment_count(self):
         Since we are supporting more or less "all" languages, it doesn't seem like we really want to count chars,
         and that counting bytes should suffice.
         """
-        content_len = len(self.content_with_placeholders_filled_in.encode("utf8"))
-        return math.ceil(content_len / 140)
+
+        # check if all chars are in the GSM-7 character set
+        def gsm_check(x):
+            rule = re.compile(r'^[\sa-zA-Z0-9_@?£!1$"¥#è?¤é%ù&ì\\ò(Ç)*:Ø+;ÄäøÆ,<LÖlöæ\-=ÑñÅß.>ÜüåÉ/§à¡¿\']+$')
+            if not rule.search(x):
+                return False
+            else:
+                return True
+
+        message_str = self.content_with_placeholders_filled_in
+
+        content_len = len(message_str)
+
+        """
+        Checks for GSM-7 char set, calculates msg size, and
+        then fragments based on multipart message rules. ASCII
+        was not specifically called out as almost all messages will
+        switch from 7bit GSM to Unicode.
```

Comment:
These calculations are based on this calculator, correct? https://messente.com/documentation/tools/sms-length-calculator

If so, I'd just link to it here directly in the comment so we have a quick reference back to it in the future, e.g.:

```suggestion
        switch from 7bit GSM to Unicode.
        
        Calculations are based on https://messente.com/documentation/tools/sms-length-calculator
```